drivers/usbhost/usbhost_hub.c

```diff
@@ -538,42 +539,51 @@ static inline int usbhost_hubdesc(FAR struct usbhost_class_s *hubclass)
   usbhost_putle16(ctrlreq->index, 0);
   usbhost_putle16(ctrlreq->len, USB_SIZEOF_HUBDESC);
 
+  ret = DRVR_ALLOC(hport->drvr, (FAR uint8_t **)&hubdesc, &maxlen);
+  if (ret < 0)
+    {
+      uerr("ERROR: DRVR_ALLOC failed: %d\n", ret);
+      return ret;
+    }
+
   ret = DRVR_CTRLIN(hport->drvr, hport->ep0,
-                    ctrlreq, (FAR uint8_t *)&hubdesc);
+                    ctrlreq, (FAR uint8_t *)hubdesc);
   if (ret < 0)
     {
       uerr("ERROR: Failed to read hub descriptor: %d\n", ret);
```

Comment:
free memory

Reply:
> free memory

line 586, free hubdesc.

Reply:
but you return at 554

Reply:
> but you return at 554

Exactly, I add hubdesc free if DRVR_CTRLIN failed.